api/room/schema.py

```diff
@@ -32,20 +33,21 @@ def mutate(self, info, **kwargs):
 class UpdateRoom(graphene.Mutation):
 
     class Arguments:
-        name = graphene.String()
+        id = graphene.Int()
         room_type = graphene.String()
         capacity = graphene.Int()
-        new_name = graphene.String()
+        name = graphene.String()
         
 
     room = graphene.Field(Room)
     
-    def mutate(self, info,name, **kwargs):
+    def mutate(self, info, id, **kwargs):
+        validate_empty_fields(**kwargs)
         query_room = Room.get_query(info)
-        exact_room = query_room.filter(RoomModel.name == name).first()
+        exact_room = query_room.filter(RoomModel.id == id).first()
 
-        if kwargs.get("new_name"):
-            exact_room.name = kwargs["new_name"]
+        if kwargs.get("name"):  
```

Comment:
We could get cleaner code if we just spread the kwargs and supplied them as fields to the object, this can be achieved by supplying them to the save method in the format save(**kwargs), this can even be done in a helper class or we can override the save method of the object type.

Reply:
@shemogumbe, Thanks. I totally agree and moving forward code quality shall be upheld. I have refactored by introducing an `update_entity_fields` in the utilities. This method can be used for any other mutation for updating.